### ISSUE TITLE
Create templates for project instructions

### ISSUE THREAD

**Issue by geohacker** [hotosm/tasking-manager#1111]
It'd be great if we can support templates for project instructions that people can edit/fill in when they create new projects - this will make sure all the necessary instructions are captured, and people wouldn't have to copy/paste or try to come up with new text all the time.

cc @bhousel @geomantic @mcawley

**Comment by bgirardot** [comment]
Hi,

We have something like this now, the text at least has been worked out and translated.

It is more complicated that it seems from a UI perspective, because the point is to have a good, well written set of instructions translated, but you want to make any changes, you have to manage it across all the languages.

We were thinking something like a checkbox for each instruction, automatically grouped into the headings, just check which instructions you would like included. That is most of the editing, "this one does not apply to me, etc etc".

These are in markdown basically, you can copy paste them to projects now if you 'edit source' to get the markdown for the Tasking Manager (already partially translated):

https://wiki.openstreetmap.org/wiki/OSM_Tasking_Manager/Default_roads,_buildings,_waterways_instructions

Original version here (pure markdown for copy/paste):
https://wiki.openstreetmap.org/wiki/User:Bgirardot/Tasking_Manager_Default_Roads,_Buildings,_Waterways_Instructions

**Comment by xamanu** [comment]
I would like to add to this the idea of maintaining instructions on the OSM wiki, within data items, that are machine readable and can be pulled in by the Tasking Manager. This could potentially update the instructions on older projects, too.

**Comment by russdeffner** [comment]
Great to see this conversation started again. I wonder if the wiki is not the ideal place for the instructions. We need them to be very basic and applicable globally, while allowing for pm's to add the local schema. I could see something like 'standard highway instructions' to be very enticing to edit a whole bunch on the wiki and if they're automatically ported/updated into projects that may not be a good idea. I think we want some kind of 'versioning control' on default instructions so we can delicately craft them to apply to all projects of that type and make sure changes don't negatively affect existing projects.

**Comment by giblet22** [comment]
@russdeffner is there already a wiki page set up for this?

**Comment by russdeffner** [comment]
Yes @giblet22 - see Blake's post above; but I don't mind repeating myself - I do not think the wiki is the best place for these template instructions if we intend to create 'automatic' instructions for TM projects. I wonder what would prevent someone from 'hacking'/vandalizing them or generally just making changes that they think are 'correct' due to their local situation but causes "global" instructions to suddenly be incorrect. I would suggest here in Github is better, or even a Google doc so we have some versioning control and at least a one layer approval of changes.

**Comment by willemarcel** [comment]
@russdeffner @giblet22 I think GitHub is a good place, we can create a new folder to it, write in Markdown and also open the possibility to have translations.

**Comment by petya-kangalova** [comment]
@ramyaragupathy  this was brought up again in the Tasking Manager Meet Up. That is will be useful to have templates. It will be good to update the community how that fits  in the roadmap.

**Comment by ramyaragupathy** [comment]
Time to revisit this :) @prabinoid @suzit-10
